bittensor/dendrites/pkm.py

```diff
@@ -213,8 +215,8 @@ def _route(self, inputs, query, modality: bittensor_pb2.Modality) -> Tuple[torch
 
         # weights: (torch.LongTensor): weights scattered onto uids per example.
         # weights.shape = [batch_size, metagraph.n]
-        weights = torch.zeros(inputs.shape[0], self.session.metagraph.n).to(self.device)
-        weights.scatter_(1, indices.repeat(batch_size, 1), scores)
+        weights = torch.zeros(inputs.shape[0], self.session.metagraph.n)
```

Comment:
Not sure why you took out the `.to(self.device)`, this would break CUDA compatibility on the subsequent `weights.scatter_` call in the next line. Since `indices` is in the GPU and `weights` isn't.